examples/Cassie/run_pd_controller.cc

```diff
@@ -1,49 +1,79 @@
+#include <gflags/gflags.h>
+
+#include "systems/primitives/output_subvector_pass_through.h"
+#include "drake/systems/analysis/simulator.h"
 #include "drake/systems/framework/diagram_builder.h"
+#include "drake/systems/lcm/lcm_interface_system.h"
 #include "drake/systems/lcm/lcm_publisher_system.h"
 #include "drake/systems/lcm/lcm_subscriber_system.h"
-#include "drake/systems/lcm/lcm_interface_system.h"
-#include "drake/systems/analysis/simulator.h"
 
-#include "dairlib/lcmt_robot_output.hpp"
-#include "dairlib/lcmt_robot_input.hpp"
 #include "dairlib/lcmt_pd_config.hpp"
-#include "systems/robot_lcm_systems.h"
+#include "dairlib/lcmt_robot_input.hpp"
+#include "dairlib/lcmt_robot_output.hpp"
+#include "examples/Cassie/cassie_utils.h"
 #include "systems/controllers/linear_controller.h"
 #include "systems/controllers/pd_config_lcm.h"
-#include "examples/Cassie/cassie_utils.h"
+#include "systems/robot_lcm_systems.h"
 
 namespace dairlib {
 
-using drake::systems::lcm::LcmSubscriberSystem;
-using drake::systems::lcm::LcmPublisherSystem;
-using drake::multibody::MultibodyPlant;
 using drake::geometry::SceneGraph;
+using drake::multibody::MultibodyPlant;
 using drake::systems::DiagramBuilder;
+using drake::systems::lcm::LcmPublisherSystem;
+using drake::systems::lcm::LcmSubscriberSystem;
+
+DEFINE_string(channel_x, "CASSIE_STATE_DISPATCHER",
+              "LCM channel for receiving state. "
+              "Use CASSIE_STATE_SIMULATION to get state from simulator, and "
+              "use CASSIE_STATE_DISPATCHER to get state from state estimator");
 
+DEFINE_string(channel_u, "CASSIE_INPUT",
+              "The name of the channel which publishes command");
+
+int doMain(int argc, char* argv[]) {
+  gflags::ParseCommandLineFlags(&argc, &argv, true);
 
-int doMain() {
   DiagramBuilder<double> builder;
   DiagramBuilder<double> builder_null;
 
   auto lcm = builder.AddSystem<drake::systems::lcm::LcmInterfaceSystem>(
       "udpm://239.255.76.67:7667?ttl=0");
 
   SceneGraph<double>& scene_graph = *builder.AddSystem<SceneGraph>();
+  SceneGraph<double>& scene_graph_floating =
+      *builder_null.AddSystem<SceneGraph>();
   scene_graph.set_name("scene_graph");
 
-  MultibodyPlant<double>& plant =
+  MultibodyPlant<double>& plant = *builder_null.AddSystem<MultibodyPlant>(1.0);
+  MultibodyPlant<double>& plant_floating =
       *builder_null.AddSystem<MultibodyPlant>(1.0);
-  addCassieMultibody(&plant, &scene_graph, false);
+
+  addCassieMultibody(&plant, &scene_graph_floating, false);
```

Comment:
1. You are right. `floating-base` model should be an option. I've added a flag to make it optional
2. We retain the fixed-base plant because both `PDConfigReceiver` and `LinearController` use only the joint position (without the floating base coordinates)

Reply:
I'm not sure what you mean here. The PD controller should work with the floating base plant, unless I'm mistaken. It cannot control the floating base joints, but otherwise I believe it was coded to be general enough to handle unactuated DOFs.

Reply:
It doesn't look like `LinearController` ignore the under-actuated joint?
https://github.com/DAIRLab/dairlib/blob/master/systems/controllers/linear_controller.cc#L27-L28

Reply:
After discussing with @mposa , I realized that `PDConfigReceiver` already takes care of the floating base part of the model when creating the gain matrix K. Therefore there is no need for having two plants.